src/frame.rs

```diff
@@ -204,11 +206,18 @@ impl Frame {
         let mut data_vec = Vec::with_capacity(num_planes);
         let linesize = self.frame.Linesize;
 
+        // very sketchy, try to find a better way
+        let pix_fmt: AVPixelFormat =
+            unsafe { mem::transmute(self.frame.EncodedPixelFormat) };
```

Comment:
This is bad, but I know there is not another method right now. The best way would be to add into `ffmpeg-the-third` a function to convert an `i32` into an enum, as explained in this blog post https://enodev.fr/posts/rusticity-convert-an-integer-to-an-enum.html

If you want you can try to open a PR upstream, better the first option to avoid dependencies in my opinion, a function like [this](https://github.com/shssoichiro/ffmpeg-the-third/blob/fe89e158c032b677ec83c1fcdfac9819c53e3918/src/util/format/pixel.rs#L434) for `i32`

Reply:
That would be ideal. Would different enum definitions across different ffmpeg versions ([as FFMS2 API doc warns](https://github.com/FFMS/ffms2/blob/master/doc/ffms2-api.md#ffms_frame:~:text=It%20is%20strongly%20recommended%20to%20use%20this%20function%20instead%20of%20including%20pixfmt.h%20directly%2C%20since%20this%20function%20guarantees%20that%20you%20will%20always%20get%20the%20constant%20definitions%20from%20the%20version%20of%20FFmpeg%20that%20FFMS2%20was%20linked%20against.)) be an issue? I'm worried that conditions like [here](https://github.com/shssoichiro/ffmpeg-the-third/blob/fe89e158c032b677ec83c1fcdfac9819c53e3918/src/util/format/pixel.rs#L455) and [here](https://github.com/shssoichiro/ffmpeg-the-third/blob/fe89e158c032b677ec83c1fcdfac9819c53e3918/src/util/format/pixel.rs#L505) would cause i32 -> AVPixelFormat mappings to be inconsistent, which wouldn't be a problem with `mem::transmute` if we assume that ffmpeg-the-third and ffms2 use the same ffmpeg.

Reply:
Yep, you're right, but as it is now, this crate uses the last `ffms2` version, so an `ffmpeg` compatible with that version. Thus assuming that `ffmpeg-the-third` and `ffms2` have the same `ffmpeg` version is not wrong in my opinion. And a developer can re-build this crate setting up the correct `ffmpeg-the-third` version through features.

Reply:
So if a match statement for example 
```rust
match {
    1 => Pixel::YUV422P16BE
    2 => Pixel::YUV444P16BE,
    #[cfg(feature = "ff_api_vdpau")]
    3 => Pixel::VDPAU_MPEG4,
    4 => Pixel::DXVA2_VLD,
}
```
were in `ffmpeg-the-third`, would it still be valid regardless of `ff_api_vdpau` being enabled in `ffmpeg`? 
So if `ff_api_vdpau` was disabled, would `Pixel::DXVA2_VLD` still be 4, or should it actually be 3?

Reply:
If `ff_api_vdpau` is enabled, `Pixel::DXVA2_VLD` remains 4. If it's disabled, it remains 4 the same

Reply:
Oh, I should've been more clear. So [here in the `ffmpeg` source](https://github.com/FFmpeg/FFmpeg/blob/master/libavutil/pixfmt.h#L291), if the flag is disabled, all the enum values after that would have their underlying `int` representation (which `FFMS_GetPixFmt` returns) reduced by one, and therefore invalidate a hardcoded match statement to convert `i32` -> `AVPixelFormat`.

So in my previous message, if `ff_api_vdpau` was disabled in both `ffmpeg-the-third` and `ffmpeg`, would that mean 4 being mapped to `Pixel::DXVA2_VLD` is incorrect, and should somehow be the mapping for 3 instead? Similarly, all the following mapping would need to change as well.

Reply:
@shssoichiro Could you help us to unravel the tangle?

Reply:
@ben476 

Perhaps we can tackle this problem using the value of the `XVMC` field. Since it's the only preprocessor flags in `ffmpeg` we can adopt this condition.

```
if xmvc_value == "value with flags disabled" and my_enum_value > xmvc_value {
   return my_enum_value - 1
}
```
Practically we assume that the flags is always enabled

Reply:
As with anything in ffmpeg, this seems to be a shortcoming of the ffmpeg API. This was not thought through when they implemented it.

The above suggestion is probably the best option @Luni-4 unfortunately...

Reply:
I agree and it should fix the problem. What do you think @ben476?

Reply:
I'm not too sure... Previous versions of `ffmpeg` have different flag checks and have also removed flags from the middle of the enum, such as [here](https://github.com/FFmpeg/FFmpeg/commit/ad524cb9eeb298f7a60d923094fbebb1fda7e0e3)

It seems like you would need to check a lot more flags, as well as the `ffmpeg` version.

Reply:
If we decide to consider more versions, we need to support all flags of each of the considered version. So we need to add more checks and enable/disable them through rust features

Reply:
Is `transmute` still an option? Maybe making a PR in `ffms2` to expose chroma height in the frame struct would be a better option at this point

Reply:
Yeah, I agree with you. Let's keep the `transmute` option for now, but please write as comment above that line of code  all possible solutions we have though of here. And a PR in `ffms2` would be great

Reply:
Cool, I'm happy to do that comment. I'm quite busy from now on, so I won't be able to work on the PRs for a while, but I'll make sure to get to it when I have the chance.

Reply:
In fact given the fact that this isn't the only case where this happens in ffmpeg's enums, it might be better to consider a solution that doesn't involve converting the enums to/from an integer, if it's possible to convert to the enum from its stringified name instead perhaps?

Reply:
That's a good idea, but it doesn't seem that FFMS2 provides a function to convert the integer identifier (which we have) to a name, despite there being a function for the other way around.

Reply:
For now let's land this PR, waiting for a solution upstream in `ffmpeg-the-third`